sdk/identity/azure-identity/README.md

```diff
@@ -0,0 +1,186 @@
+
+SecretClient client = SecretClient.builder()
+    .endpoint("https://{YOUR_VAULT_NAME}.vault.azure.net")
+    .credential(credential)
+    .build();
+```
+When executing this in a development machine you need to first [configure the environment](#environment-variables) setting the variables `AZURE_CLIENT_ID`, `AZURE_TENANT_ID` and `AZURE_CLIENT_SECRET` to the appropriate values for your service principal.
+
+## Authenticating a service principal with a client secret
+This example demonstrates authenticating the `KeyClient` from the [azure-keyvault-keys][keys_client_library] client library using the `ClientSecretCredential`.
+```java
+// using a client secret
+import com.azure.identity.credential.ClientSecretCredential;
+import com.azure.security.keyvault.keys.KeyClient;
+
+// authenticate with client secret,
+ClientSecretCredential clientSecretCredential = new ClientSecretCredential()
+	    .clientId("<YOUR_CLIENT_ID>")
+	    .clientSecret("<YOUR_CLIENT_SECRET>")
+	    .tenantId("<YOUR_TENANT_ID>");
+
+KeyClient client = KeyClient.builder()
+    .endpoint("https://{YOUR_VAULT_NAME}.vault.azure.net")
+    .credential(clientSecretCredential)
+    .build();
+```
+
+## Chaining credentials:
+The `ChainedTokenCredential` class provides the ability to link together multiple credential instances to be tried sequentially when authenticating. The following example demonstrates creating a credential which will attempt to authenticate using managed identity, and fall back to certificate authentication if a managed identity is unavailable in the current environment. This example authenticates an `EventHubClient` from the [azure-eventhubs][eventhubs_client_library] client library using the `ChainedTokenCredential`.
+
+```java
+import com.azure.identity.credential.ClientSecretCredential;
+import com.azure.security.keyvault.secrets.SecretClient;
+
+ClientSecretcredential firstServicePrincipal = new ClientSecretCredential()
+	    .clientId("<YOUR_CLIENT_ID>")
+	    .clientSecret("<YOUR_CLIENT_SECRET>")
+	    .tenantId("<YOUR_TENANT_ID>");
+
+ClientSecretcredential secondServicePrincipal = new ClientSecretCredential()
+	    .clientId("<YOUR_CLIENT_ID>")
+	    .clientSecret("<YOUR_CLIENT_SECRET>")
+	    .tenantId("<YOUR_TENANT_ID>");
+
+// when an access token is requested, the chain will try each
+// credential in order, stopping when one provides a token
+
+ChainedTokenCredential credentialChain = new ChainedTokenCredential()
+		.addLast(firstServicePrincipal)
+		.addLast(secondServicePrincipal);
+
+// the chain can be used anywhere a credential is required
+
+// The fully qualified host name for the Event Hubs namespace. This is likely to be similar to:
+// {your-namespace}.servicebus.windows.net
+String host = "<< EVENT HUBS HOST >>"
+String eventHubPath = "<< NAME OF THE EVENT HUB >>";
+EventHubClient client = new EventHubClientBuilder()
+    .credential(host, eventHubPath, credentialChain)
+    .build();
```

Comment:
This is buildAsyncClient() now I believe.

Reply:
updated